app/javascript/components/AddLocation.jsx

```diff
@@ -0,0 +1,83 @@
+import React from "react"
+import PropTypes from "prop-types"
+class AddLocation extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      buttonActive: false,
+      name: null
+    }
+  }
+
+  guid = () => {
+    let s4 = () => {
+      return Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
+    }
+    return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
+      s4() + '-' + s4() + s4() + s4();
+  }
+
+  addLocationBox = (e) => {
+    if(this.state.buttonActive) {
+      let id = this.guid();
+      this.props.addLocationHandler(e, {
+        admin_url: null,
+        height: 50,
```

Comment:
A little reordering here to keep like values together?
i.e width,
     height,
     position_x,
     position_y